File: sdks/go/tests/conformance/client_creation_test.go
package conformance

import (
	"context"
	"testing"
	"time"

	"github.com/astraive/loza/sdks/go"
)

func TestClientCreationAndConfiguration(t *testing.T) {
	sink, store := loza.MemorySink()
	cfg := loza.ApplyConfig(
		loza.Test(),
		loza.WithService("verification"),
		loza.WithEnvironment("prod"),
		loza.WithRelease("1.2.3"),
		loza.WithNamespace("payments"),
		loza.WithAPIKey("secret"),
		loza.WithOtelBridge(true),
		loza.WithRetry(2),
		loza.WithTimeout(2*time.Second),
		loza.WithQueueSize(128),
		loza.WithSink(sink),
	)

	logger, err := loza.CreateLoza(cfg)
	if err != nil {
		t.Fatalf("create logger: %v", err)
	}
	if logger.Config().Service != "verification" {
		t.Fatalf("expected service verification, got %q", logger.Config().Service)
	}
	if err := loza.Configure(cfg); err != nil {
		t.Fatalf("configure default: %v", err)
	}
	aliased, err := loza.Alias("audit")
	if err != nil {
		t.Fatalf("alias: %v", err)
	}
	if aliased.Config().Alias != "audit" {
		t.Fatalf("expected alias audit, got %q", aliased.Config().Alias)
	}
	if logger.Config().Alias != "" {
		t.Fatalf("expected parent logger alias to remain empty")
	}
	logger.Info("configured logger", loza.String("family", "config"))
	if err := logger.Flush(context.Background()); err != nil {
		t.Fatalf("flush configured logger: %v", err)
	}
	if store.Len() == 0 {
		t.Fatalf("expected configured logger to emit an event")
	}
}

File: sdks/js/tests/client_creation.test.ts
import { afterEach, it } from 'node:test';
import assert from 'node:assert/strict';
import * as loza from '../src/index.ts';

afterEach(() => {
  loza.reset();
});

it('covers client creation, configuration, and aliasing', async () => {
  const sink = new loza.MemorySink();
  loza.configure(loza.production('catalog').withSink(sink));

  const logger = loza.createLoza({ service: 'catalog-child', sink: new loza.MemorySink() });
  assert.equal(logger.getConfig().service, 'catalog-child');

  const builder = loza
    .test('cfg')
    .withRelease('2026.05.24')
    .withNamespace('payments');
  const built = builder.build();
  assert.equal(built.release, '2026.05.24');
  assert.equal(built.namespace, 'payments');

  const aliased = loza.alias('audit');
  assert.equal(aliased.getConfig().service, 'catalog');
  assert.equal(aliased.getConfig().alias, 'audit');

  await logger.shutdown();
});
